RecoTracker/LSTCore/src/alpaka/Quadruplet.h

```diff
@@ -0,0 +1,1065 @@
+                                                      mds,
+                                                      segments,
+                                                      triplets,
+                                                      lowerModule1,
+                                                      lowerModule2,
+                                                      lowerModule3,
+                                                      lowerModule4,
+                                                      innerTripletIndex,
+                                                      outerTripletIndex,
+                                                      regressionG,
+                                                      regressionF,
+                                                      regressionRadius,
+                                                      nonAnchorRegressionRadius,
+                                                      chiSquared,
+                                                      ptCut,
+                                                      rzChiSquared,
+                                                      nonAnchorChiSquared,
+                                                      dBeta,
+                                                      promptScore,
+                                                      displacedScore,
+                                                      fakeScore);
+              if (success) {
+                int totOccupancyQuadruplets =
+                    alpaka::atomicAdd(acc,
+                                      &quadrupletsOccupancy.totOccupancyQuadruplets()[lowerModule1],
+                                      1u,
+                                      alpaka::hierarchy::Threads{});
+                if (totOccupancyQuadruplets >= ranges.quadrupletModuleOccupancy()[lowerModule1]) {
+#ifdef WARNINGS
+                  printf("Quadruplet excess alert! Module index = %d, Occupancy = %d\n",
+                         lowerModule1,
+                         totOccupancyQuadruplets);
+#endif
+                } else {
+                  int quadrupletModuleIndex = alpaka::atomicAdd(
+                      acc, &quadrupletsOccupancy.nQuadruplets()[lowerModule1], 1u, alpaka::hierarchy::Threads{});
+                  //this if statement should never get executed!
+                  if (ranges.quadrupletModuleIndices()[lowerModule1] == -1) {
+#ifdef WARNINGS
+                    printf("Quadruplets : no memory for module at module index = %d\n", lowerModule1);
+#endif
+                  } else {
+                    unsigned int quadrupletIndex =
+                        ranges.quadrupletModuleIndices()[lowerModule1] + quadrupletModuleIndex;
+                    float phi =
+                        mds.anchorPhi()[segments.mdIndices()[triplets.segmentIndices()[innerTripletIndex][md_adjustment]]
+                                                            [layer2_adjustment]];  //layer 3
```

Comment:
at least for readability pull the repeated index call out to a temp

```suggestion
                    auto mdLayer3Index = triplets.segmentIndices()[innerTripletIndex][md_adjustment]][layer2_adjustment];
                    float phi = mds.anchorPhi()[mdLayer3Index];
```